src/complex/Utilities/OStreamUtilities.cpp

```diff
@@ -0,0 +1,968 @@
+                                                                  const IFilter::MessageHandler& mesgHandler, const std::string& delimiter, bool includeIndex, size_t componentsPerLine,
+                                                                  bool includeHeaders, bool includeNeighborLists)
+{
+  bool hasNeighborLists = false;
+  auto objTypes = getDataTypesWrapper(objectPaths, dataStructure);
+  for(const auto& objType : objTypes)
+  {
+    if(objType == DataObject::Type::NeighborList)
+    {
+      hasNeighborLists = true;
+    }
+  }
+  auto sortedMap = std::move(createSortedMapbyType(objectPaths, objTypes));
+  auto matrices = unpackSortedMapIntoMatricies(sortedMap, dataStructure, mesgHandler, includeIndex, includeHeaders, includeNeighborLists);
+
+  ParallelDataAlgorithm dataAlg;
+  std::vector<std::map<size_t, std::string>> stringStoreList;
+  auto count = 0;
+  for(auto& matrix : matrices) // neighborLists automatically stored at the end
+  {
+    mesgHandler(IFilter::Message::Type::Progress, fmt::format("Now processing print object {} of {}", count, matrices.size()));
+    std::vector<std::string> stringStore(matrix->getSize(), "UNINITIALIZED"); // 1 per matrix
+    dataAlg.setRange(0, matrix->getSize());
+    dataAlg.execute(OStreamUtilities::AssembleHorizontalStringFromIndex(*matrix, stringStore, delimiter, componentsPerLine));
+    stringStoreList.emplace_back(createStringMapFromVector(stringStore));
+    count++;
+  }
+
+  mesgHandler(IFilter::Message::Type::Info, "Printing out");
+  for(size_t i = 0; i < stringStoreList.size(); i++)
+  {
+    writeOutWrapper(stringStoreList[i], outputStrm);
+    if(i < (stringStoreList.size() - 1))
+    {
+      outputStrm << "\n"; // seperate neighborlists
+    }
+  }
+}
+
+// Creates OFStream from filepath [NO BINARY SUPPORT]
+void OStreamUtilities::OutputFunctions::printDataSetsToSingleFile(const std::vector<DataPath>& objectPaths, DataStructure& dataStructure, std::filesystem::path& filePath,
+                                                                  const IFilter::MessageHandler& mesgHandler, const std::string& delimiter, bool includeIndex, size_t componentsPerLine,
+                                                                  bool includeHeaders, bool includeNeighborLists)
+{
+  bool hasNeighborLists = false;
+  auto objTypes = getDataTypesWrapper(objectPaths, dataStructure);
+  for(const auto& objType : objTypes)
+  {
+    if(objType == DataObject::Type::NeighborList)
+    {
+      hasNeighborLists = true;
+    }
+  }
+  auto sortedMap = std::move(createSortedMapbyType(objectPaths, objTypes));
+  auto matrices = unpackSortedMapIntoMatricies(sortedMap, dataStructure, mesgHandler, includeIndex, includeHeaders, includeNeighborLists);
+
+  ParallelDataAlgorithm dataAlg;
+
+  std::vector<std::map<size_t, std::string>> stringStoreList;
+  auto count = 0;
```

Comment:
```suggestion
  usize count = 0;
```
`auto` with literals can be tricky. In this case, the literal is `int`, but since since this is a count for `std::vector`, `size_t`/`usize` is more appropriate.